### PR TITLE
Add profile updater request

### PR DESCRIPTION
**Pull request by friedger**
## Description

As a user, I want to update my public profile from owl.link or similar apps that manage some profile data for me. As an app developer, I want to update the user's public profile.

This PR adds a request to connect for updating the user's profile.

This is part of https://github.com/hirosystems/stacks-wallet-web/issues/2544

## Type of Change
- [x] New feature
- [ ] Bug fix
- [ ] API reference/documentation update
- [ ] Other

## Does this introduce a breaking change?
No

## Are documentation updates required?
Not sure where the protocol is documented.## Checklist
- [x] Code is commented where needed
- [ ] Unit test coverage for new or modified code paths
- [ ] `yarn lerna run test` passes
- [ ] Changelog is updated
- [ ] Tag 1 of @kyranjamie or @aulneau for review

### CODE REVIEW

**Review by janniks — COMMENTED** [review]
In general, this looks good. _(I can take a closer look over the next week)_

If we want to launch this feature, we should coordinate and test closely with the wallet. Because, this basically need to go live at the same time as the wallets part (otherwise it might be confusing for devs that one part exists, but not the other half)...

For now, I created a fork of this branch (to merge back from/to), with some minor edits.
The branch is auto-deployed to npm under `5.0.6-alpha.13dd3b3.0` (use `--save-exact`, when installing). You can use that version for development and add tests (and ideally an implementation of how apps will use this to the `test-app` inside the wallet repo).

**Comment by friedger** [comment]
@janniks Thank you. Very helpful!

File: packages/connect/src/types/profileUpdater.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+}
+
+export interface CommonProfileUpdaterRequestOptions extends ProfileUpdaterBase {}
+
+export interface CommoneProfileUpdaterPayload {
+  publicKey: string;
+}
+
+export interface PersonProfileUpdaterPayload extends CommoneProfileUpdaterPayload {
+  profile: Person;
```
Comment (friedger) [inline packages/connect/src/types/profileUpdater.ts:26]: @kyranjamie This should be just a type, not a class, correct?

Comment (kyranjamie) [reply packages/connect/src/types/profileUpdater.ts:26]: If it's passed to the wallet, then yes

**Comment by friedger** [comment]
This is replaced by #279

**Pull request closed without merging**